### PR TITLE
add api call for equipment types, multi select component for user to …

### PR DESCRIPTION
**Pull request by chrismcanulty**
…select available equipment and built skeleton for desired number of exercises

JIRA ticket: https://lhl-midterm-ecommerce.atlassian.net/browse/WG-10

Pending work: add logic to ensure user has selected at least one item from multi select component

Screenshots

Screen Shot 2023-03-30 at 8 20 27 PM

Screen Shot 2023-03-30 at 8 20 37 PM

Screen Shot 2023-03-30 at 8 20 43 PM

### CODE REVIEW

File: src/screens/Equipment.tsx [hunk 1]
```diff
@@ -1,17 +1,75 @@
...
+
+  const equipmentData = () => {
+    const temp = [];
+    for (const equipment of equipmentTypes as any) {
+      if (equipment.name) {
+        temp.push({...equipment, key: equipment.id, value: equipment.name});
+      }
+    }
+    return temp;
+  };
```
Comment (wadesun18) [inline src/screens/Equipment.tsx:60]: I’d recommend that you do this in user context so it’s cleaner

File: src/screens/Equipment.tsx [hunk 2]
```diff
@@ -1,17 +1,75 @@
...
   return (
     <View>
       <Header>{questions[1]}</Header>
+      <MultipleSelectList
+        setSelected={setSelectedEquipment}
+        data={equipmentData()}
+        save="value"
+        boxStyles={{marginLeft: 20, marginRight: 20, borderRadius: 20}}
+        dropdownStyles={{marginLeft: 20, marginRight: 20, borderRadius: 20}}
+        // onSelect={selected => setSelectedMuscles(selected)}
```
Comment (wadesun18) [inline src/screens/Equipment.tsx]: Remove

File: src/screens/Equipment.tsx [hunk 3]
```diff
@@ -1,17 +1,75 @@
...
+        label="Categories"
+      />
+      <Button
+        style={{marginBottom: 0, padding: 0}}
+        onPress={() => {
+          if (selectedEquipment.length > 0) {
+            navigation.navigate('Reps');
+          } else if (selectedEquipment.length === 0) {
+            alert('Please select what equipment you have available');
+          }
```
Comment (wadesun18) [inline src/screens/Equipment.tsx]: Separate out the function

File: src/screens/Equipment.tsx [hunk 4]
```diff
@@ -1,17 +1,75 @@
...
+        }}>
+        <ButtonText style={{marginTop: 100, marginBottom: 0, padding: 0}}>
+          Next
+        </ButtonText>
+      </Button>
```
Comment (wadesun18) [inline src/screens/Equipment.tsx:80]: What is equipment type is empty

File: src/screens/HomeScreen.tsx [hunk 5]
```diff
@@ -77,7 +79,13 @@ export default function HomeScreen({navigation}: NativeStackHeaderProps) {
...
       <Button
         style={{marginBottom: 0, padding: 0}}
-        onPress={() => navigation.navigate('Equipment')}>
+        onPress={() => {
+          if (selectedMuscles.length > 0) {
+            navigation.navigate('Equipment');
+          } else if (selectedMuscles.length === 0) {
+            alert('Please select at least one muscle group');
+          }
+        }}>
```
Comment (wadesun18) [inline src/screens/HomeScreen.tsx]: Separate out the function

**Review by wadesun18 — CHANGES_REQUESTED** [review]

**Comment by chrismcanulty** [comment]
Proposed changes to React Native Dropdown Select List open source code in bold **

Link: https://www.npmjs.com/package/react-native-dropdown-select-list

export const MultipleSelectList: React.FC<MultipleSelectListProps> = ({
  fontFamily,
  setSelected,
  placeholder,
  boxStyles,
  inputStyles,
  dropdownStyles,
  dropdownItemStyles,
  dropdownTextStyles,
  maxHeight,
  data,
  searchicon = false,
  arrowicon = false,
  closeicon = false,
  search = true,
  searchPlaceholder = 'search',
  onSelect = () => {},
  label,
  notFoundText = 'No data found',
  disabledItemStyles,
  disabledTextStyles,
  disabledCheckBoxStyles,
  labelStyles,
  badgeStyles,
  badgeTextStyles,
  checkBoxStyles,
  save = 'key',
  dropdownShown = false,
  **defaultValues,**
}) => {
  const oldOption = React.useRef(null);
  const [_firstRender, _setFirstRender] = React.useState<boolean>(true);
  const [dropdown, setDropdown] = React.useState<boolean>(dropdownShown);
  const [selectedval, setSelectedVal] = React.useState<any>(
    **defaultValues || []**,
  );

File: src/screens/HomeScreen.tsx [hunk 6]
```diff
@@ -49,7 +49,7 @@ export default function HomeScreen({navigation}: NativeStackHeaderProps) {
     useContext(UserContext);
   const [warning, setWarning] = useState(false);
 
-  if (muscleGroup.length === 0) return null;
+  // if (muscleGroup.length === 0) return null;
```
Comment (wadesun18) [inline src/screens/HomeScreen.tsx:52]: Remove

**Review by wadesun18 — APPROVED** [review]

**Pull request merged**